mlflow/openai/__init__.py

```diff
@@ -576,22 +576,66 @@ def _has_content_and_role(d):
     return "content" in d and "role" in d
 
 
-class _FormattableMessage:
-    def __init__(self, message):
-        self.content = message.get("content")
-        self.role = message.get("role")
-        self.variables = _parse_format_fields(self.content)
+class _ContentFormatter:
+    def __init__(self, type, template=None):
+        if type == "completions":
+            if not template:
+                template = "{prompt}"
+            if not isinstance(template, str):
+                raise mlflow.MlflowException.invalid_parameter_value(
+                    f"Type {type(template)} is not a valid type for template of task {type}."
```

Comment:
tiny nit: can we alert that we're expecting a str at the end of the exception message?

Reply:
Fixed!